### PR TITLE
FEATURE BRANCH: thuang-837-process-status

### PR DESCRIPTION
**Pull request by tihuan**
#837 
MAKE THIS THE FEATURE BRANCH!

#### Reviewers
**Functional:** 
@Bento007 

**Readability:** 
@seve 
---

## Changes
- add
  - `PROCESSING_STATUS` enum
  - Use such enum as a new failure scenario
- modify
  - `useCheckCollection()` should check for `isFailed` to prevent permanent fetching
 
When we get `processing_status === PROCESSING_STATUS.FAILURE` we show the following message:

Screen Shot 2021-01-25 at 4 59 49 PM

### CODE REVIEW

File: frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts [hunk 1]
```diff
@@ -16,15 +17,28 @@ export enum FAILED_RETURN_TYPE {
...
 
 export function checkIfFailed(datasetStatus: DatasetUploadStatus): FailReturn {
+  if (datasetStatus.processing_status === PROCESSING_STATUS.FAILURE) {
+    return {
+      error: PROCESSING_STATUS.FAILURE,
+      isFailed: true,
+      type: FAILED_RETURN_TYPE.PROCESS,
+    };
+  }
+
```
Comment (seve) [inline frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts:41]: If the processing status failure's message is vague, do we want to prioritize others in the case of multiple failures?

Comment (tihuan) [reply frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts]: Ooh that's a really good point 😆 Thanks for pointing that out, Seve! 🙏 DONE!

**Review by seve — APPROVED** [review]

**Comment by tihuan** [comment]
Thanks @seve ! We just need @Bento007 's help to fix the unit test in [this PR](https://github.com/chanzuckerberg/corpora-data-portal/pull/867), then we can merge his branch into this feature branch, and merge all into `main` :fire!

**Review by Bento007 — APPROVED** [review]
LGTM

**Comment by tihuan** [comment]
rerunning a failed unit test! I think it's a flaky one 👌

**Pull request merged**